brainscore/benchmarks/moeller2017.py

```diff
@@ -0,0 +1,503 @@
+        }
+
+        assembly = self._load_assembly()
+
+        target_stimulus_class = STIMULUS_CLASS_DICT[self._stimulus_class]
+        target_assembly = assembly.multisel(object_name=target_stimulus_class)  # TODO
+        # TODO add object_ID & image_ID coords; image_ID = expression or viewing angle
+        return target_assembly
+
+    @staticmethod
+    def _load_assembly():
+        '''
+        TODO Load DataArray from path
+        TODO package all data
+        TODO get stimulus set
+        make sure it has a monkey dimension for ceiling computation
+        :return: DataArray
+        '''
+        assembly = None  # TODO
+        return assembly
+
+
+def Moeller2017Exp1():
+    '''
+    EXP1:
+    Stimulate face patch during face identification
+    '''
+    return _Moeller2017(stimulus_class='faces',
+                        perturbation_location='within_facepatch',
+                        identifier='dicarlo.Moeller2017-Experiment_1')
+
+
+def Moeller2017Exp2():
+    '''
+    TODO  very unclean
+    EXP2:
+    i: Stimulate outside of the face patch during face identification
+    ii: Stimulate face patch during object identification
+    '''
+
+    class _Moeller2017Exp2(BenchmarkBase):
+        def __init__(self):
+            super().__init__(
+                identifier='dicarlo.Moeller2017-Experiment_2', ceiling_func=None, version=1, parent='IT', bibtex=BIBTEX)
+            self.benchmark1 = _Moeller2017(stimulus_class='faces', perturbation_location='outside_facepatch',
+                                           identifier='dicarlo.Moeller2017-Experiment_2i')
+            self.benchmark2 = _Moeller2017(stimulus_class='objects', perturbation_location='within_facepatch',
+                                           identifier='dicarlo.Moeller2017-Experiment_2ii')
+
+        def __call__(self, candidate):
+            return self.benchmark1(candidate), self.benchmark2(candidate)
+
+    return _Moeller2017Exp2()
+
+
+def Moeller2017Exp3():
+    '''
+    EXP3:
+    Stimulate face patch during face & non-face object eliciting patch response identification
+    '''
+    return _Moeller2017(stimulus_class='non_face_objects_eliciting_FP_response_plus_faces',
+                        perturbation_location='within_facepatch',
+                        identifier='dicarlo.Moeller2017-Experiment_3')
+
+
+def Moeller2017Exp4a():
+    '''
+    EXP4a:
+    Stimulate face patch during abstract face identification
+    '''
+    return _Moeller2017(stimulus_class='abstract_faces',
+                        perturbation_location='within_facepatch',
+                        identifier='dicarlo.Moeller2017-Experiment_4a')
+
+
+def Moeller2017Exp4b():
+    '''
+    EXP4b:
+    Stimulate face patch during face & abstract houses identification
+    '''
+    return _Moeller2017(stimulus_class='abstract_houses',
+                        perturbation_location='within_facepatch',
+                        identifier='dicarlo.Moeller2017-Experiment_4b')

```

Comment:
I like the way you re-use code here, as far as I can tell the implementation is quite clean in that regard. In particular, the `_Moeller2017` class has no knowledge of particular experiments, but can deal with arbitrary combinations of stimulus class and perturbations. The experiments then are the instantiation of those combinations, nice 👍